CreeDictionary/API/search/types.py

```diff
@@ -138,86 +82,90 @@ def linguistic_tag_from_fst_tags(tags: Tuple[FSTTag, ...]) -> LinguisticTag:
         return CompoundLinguisticTag(tags)
 
 
-@attrs(auto_attribs=True, frozen=True)  # frozen makes it hashable
-class SearchResult:
+class Language(Enum):
+    SOURCE = "Source"
+    TARGET = "Target"
+
+
+def wordforms_match(w1: Wordform, w2: Wordform) -> bool:
+    """Return whether two wordform objects represent the same wordform
+
+    Either they both have IDs which match, or the text and analysis match.
     """
-    Contains all of the information needed to display a search result.
+    if w1.id is not None or w2.id is not None:
+        return w1.id == w2.id
+    return w1.text == w2.text and w1.analysis == w2.analysis
+
 
-    Comment:
-    Each instance corresponds visually to one card shown on the interface
+@dataclass
+class Result:
     """
+    A target-language wordform and the features that link it to a query.
 
-    # the text of the match
-    matched_cree: str
+    The features—is it an exact source-language match for the query? Is the edit
+    distance low? &c.—that make this a candidate result for a search query are
+    required to rank different results before showing them to the user.
```

Comment:
I'm having a hard time following what this is saying

Reply:
Is this newer version more understandable, or does it need more work?


    """
    A target-language wordform and the features that link it to a query.

    Features of a wordform allow better inferences about how good a result is.

    Some examples of these features could include:
      - Is this Result a match between a source-language query and wordform
        text, or between a target-language query term and a word in the
        definition text?
      - What is the edit distance between the query term and the wordform?
      - What is the best cosine vector distance between a definition of the
        result wordform and the query terms?

    The best search results will presumably have better scores on more features
    of greater importance.

    Search methods may generate candidate results that are ultimately not sent
    to users, so any user-friendly tagging/relabelling is instead done in
    PresentationResult class.
    """

Reply:
This is much clearer to me 😄